cleanup: f-string formatting

zabop opened the pull request:
As discussed [here](https://github.com/googleapis/python-storage/issues/785), code in this repo appears directly in the documentation. (An [example](https://cloud.google.com/storage/docs/creating-buckets#prereq-code-samples), there are hundreds.) This commit introduces f-strings to increase readability of both code and documentation.

Fixes #785 🦕

zabop committed:
cleanup: f-string formatting

Mariatta commented on docs/snippets.py:
```diff
@@ -301,13 +299,13 @@ def main():
     for example in _find_examples():
         to_delete = []
-        print("%-25s: %s" % _name_and_doc(example))
+        print(f"%-25s: {_name_and_doc(example)}")
```
The conversion here doesn't look quite right.
`%-25s` adds 25 space padding to the first item from the tuple. 
Suggested alternative:
```
name, doc: _name_and_doc(example)
print(f"{name:>25}: {doc}"
```

zabop commented:
I am not sure why 2 tests failed, let me know if further input is needed from me.

Mariatta commented on samples/snippets/notification_polling.py:
```diff
@@ -76,9 +76,7 @@ def summarize(message):
-        description += "\tOverwrote generation: %s\n" % (
-            attributes["overwroteGeneration"]
-        )
+        description += f"\tOverwrote generation: {attributes['overwroteGeneration']}\n"
```
Looks like you missed converting the one below (line 83) :)

Mariatta commented on tests/unit/test_blob.py:
```diff
@@ -2875,9 +2873,9 @@ def _do_resumable_upload_call1(
         # Second mock transport.request() does sends first chunk.
         if size is None:
-            content_range = "bytes 0-{:d}/*".format(blob.chunk_size - 1)
+            content_range = f"bytes 0-{blob.chunk_size - 1:d}/*"
```
I think the `:d` format specifier is not needed anymore

```suggestion
            content_range = f"bytes 0-{blob.chunk_size - 1:}/*"
```

Mariatta commented on tests/unit/test_blob.py:
```diff
@@ -2875,9 +2873,9 @@ def _do_resumable_upload_call1(
+            content_range = f"bytes 0-{blob.chunk_size - 1:d}/*"
         else:
-            content_range = "bytes 0-{:d}/{:d}".format(blob.chunk_size - 1, size)
+            content_range = f"bytes 0-{blob.chunk_size - 1:d}/{size:d}"
```
```suggestion
            content_range = f"bytes 0-{blob.chunk_size - 1}/{size}"
```

Breathtender commented:
Yep, for the conventionalcommits message, you just need to amend the commit message to be "cleanup: f-string formatting". I thought editing the title there would do it, but seems not.
For the other one, you can run `nox -s blacken lint` and it'll lint cleanup all the files.
If you're not able to do either of those, let me know and I can help.

Mariatta reviewed (changes requested):
Thank you. Mostly looks correct, except for a few spots.

zabop commented:
@Mariatta, thank you, great catches!
@Breathtender, I get an error with `nox`:

```
% nox -s blacken lint
nox > Running session blacken
nox > Session blacken skipped: Python interpreter 3.8 not found.
nox > Running session lint
nox > Session lint skipped: Python interpreter 3.8 not found.
nox > Ran multiple sessions:
nox > * blacken: skipped
nox > * lint: skipped

```

Which I do not yet know how to solve. my Python interpreter seems to be Python 3.8 (screenshot of pottom right corner of PyCharm):

<img width="718" alt="image" src="https://user-images.githubusercontent.com/20886570/168361304-59ec5f82-c589-4090-8cb1-63b8481288f0.png">

Breathtender commented:
Oh that's strange. I'll check on why that's happening. For now I'll just run the lint myself and add the commit, once you push up your changes to address @Mariatta's comments.

zabop committed:
cleanup: f-string formatting

zabop commented:
@Breathtender, just read your reply now, sorry; in the meantime I figured out what's going on and were able to run `nox -s blacken lint`. Just pushed the results.

zabop commented:
Addressing @Mariatta suggestions now...

zabop committed:
remove unnecessary :d specifier

Co-authored-by: Mariatta Wijaya <Mariatta@users.noreply.github.com>

zabop committed:
remove unnecessary :d specifier

Co-authored-by: Mariatta Wijaya <Mariatta@users.noreply.github.com>

zabop committed:
cleanup: f-string formatting

zabop committed:
cleanup: f-string formatting

zabop commented:
I think I have addressed the suggestions.

My last commit was the result of a typo & forgetting to run `nox -s blacken lint` before. Do I have to keep the `cleanup: f-string formatting` commit messages when I push fixes to an already open PR, or is more freedom allowed here? (To rephrase the question: which commit message is checked by the conventionalcommits test?) I used the `remove unnecessary :d specifier` commit message to approve @Mariatta suggestions, I hope that's ok.

Breathtender commented:
No, it'll automatically roll all your commits together, and so the first commit message is ultimately the one that matters.

Breathtender commented:
Thanks for making those adjustments! I'll wait for @Mariatta's approval on the changes requested, but this is looking good to me.

Mariatta reviewed (approved):
Thanks! LGTM
